### PR TITLE
my blog post

### PR DESCRIPTION
**Pull request by kevinhodges**

### CODE REVIEW

File: _posts/2015-03-20-5-signs-youre-a-good-coder.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+##### 3. You spend a good percentage of your timing designing your solution.
+
+You can write all the crap code you like but if it's organised, decoupled and properly thought out, chances are it'll still be useful, maintainable and scalable (who knew!). Think about variable and function names, they're the contract you will live by for the lifetime of the system and often, coming up with a good name will help tell you when your architecture is wrong.  Think small defined units of work and how they fit together.
+
+##### 4. You dont test your own code.
+
+We're talking unit tests here.  It's fairly easy to figure out when you have finished solving a problem, but this does now mean you are no position to rationalise about the solution. You now have preconceptions and blindspots and it's very difficult to put your assumptions aside, this leads to poor tests even if coverage is high.  Do yourself and favour and let someone else write the tests for you.  Ofcourse, if you are lucky enough to be afforded TDD, this point is completely irrelevant.
```
Comment (robhuzzey) [inline _posts/2015-03-20-5-signs-youre-a-good-coder.md]: "this does now mean you are no position to rationalise about the solution" I think it's missing an "in" here?

File: _posts/2015-03-20-5-signs-youre-a-good-coder.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+title: 5 signs you're a good coder
+categories: programming
+author: kevin_hodges
+tags: software engineer coding programmer
+comments: true
+---
+
+##### 1. You don't solve problems until you understand the system/context.
+
+If this sounds like a simple one then that's because it is. You can't consistently and reliably fix problems if you don't know what else you are affecting further up or down the stack.  Take your time, understand all the local variables and touch points, pick through the flow of the system learning where data comes from and where it goes to.  In a mature system, chances are your problem has already been solved elsewhere, if others have architected their code properly you could well find your solution already exists in tools code or common functionality.
```
Comment (lukefrake) [inline _posts/2015-03-20-5-signs-youre-a-good-coder.md]: You have a few double spaces in here and dotted around the post

Comment (lukefrake) [inline _posts/2015-03-20-5-signs-youre-a-good-coder.md]: "in tools, code or common functionality" > "in tools code or common functionality"

File: _posts/2015-03-20-5-signs-youre-a-good-coder.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+##### 2. You KISS.
+
+[Keep it simple, stupid](http://en.wikipedia.org/wiki/KISS_principle).  Anybody can make a solution complicated but it takes a clever person to make a problem simple.  Regardless of the size of the problem, design your solution away from your screen, you'll be amazed at what happens when you stop thinking in terms of code and actually consider the problem as a thing. Also, it's a simple fact, the less code you write, the less chances there are for bugs to occur.
```
Comment (lukefrake) [inline _posts/2015-03-20-5-signs-youre-a-good-coder.md]: I think you should introduce KISS like this

[Keep it simple, stupid](http://en.wikipedia.org/wiki/KISS_principle) (KISS).

**Comment by robhuzzey** [comment]
Looks good to me, few comments but overall a good read. thanks.

File: _posts/2015-03-20-5-signs-youre-a-good-coder.md (hunk 3 above)
Comment (lukefrake) [inline _posts/2015-03-20-5-signs-youre-a-good-coder.md]: 'Also, it's a simple fact, the less code you write the less chances there are for bugs to occur' > 'Also, it's a simple fact, the less code you write, the less chances there are for bugs to occur'

File: _posts/2015-03-20-5-signs-youre-a-good-coder.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+If this sounds like a simple one then that's because it is. You can't consistently and reliably fix problems if you don't know what else you are affecting further up or down the stack.  Take your time, understand all the local variables and touch points, pick through the flow of the system learning where data comes from and where it goes to.  In a mature system, chances are your problem has already been solved elsewhere, if others have architected their code properly you could well find your solution already exists in tools code or common functionality.
+
+##### 2. You KISS.
+
+[Keep it simple, stupid](http://en.wikipedia.org/wiki/KISS_principle).  Anybody can make a solution complicated but it takes a clever person to make a problem simple.  Regardless of the size of the problem, design your solution away from your screen, you'll be amazed at what happens when you stop thinking in terms of code and actually consider the problem as a thing. Also, it's a simple fact, the less code you write, the less chances there are for bugs to occur.  
+
+##### 3. You spend a good percentage of your timing designing your solution.
```
Comment (lukefrake) [inline _posts/2015-03-20-5-signs-youre-a-good-coder.md]: 'your time designing' > 'your timing designing'

File: _posts/2015-03-20-5-signs-youre-a-good-coder.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+You can write all the crap code you like but if it's organised, decoupled and properly thought out, chances are it'll still be useful, maintainable and scalable (who knew!). Think about variable and function names, they're the contract you will live by for the lifetime of the system and often, coming up with a good name will help tell you when your architecture is wrong.  Think small defined units of work and how they fit together.
+
+##### 4. You dont test your own code.
+
+We're talking unit tests here.  It's fairly easy to figure out when you have finished solving a problem, but this does now mean you are no position to rationalise about the solution. You now have preconceptions and blindspots and it's very difficult to put your assumptions aside, this leads to poor tests even if coverage is high.  Do yourself and favour and let someone else write the tests for you.  Ofcourse, if you are lucky enough to be afforded TDD, this point is completely irrelevant.
+
+##### 5. You've learnt to detect code smell and more importantly, you act on it.
```
Comment (lukefrake) [inline _posts/2015-03-20-5-signs-youre-a-good-coder.md]: Maybe 'You've got a keen nose for code smell' to keep the metaphor going

**Comment by lukefrake** [comment]
Really good post mate, a couple of points but they are mostly more opinions than anything else.

Might be nice to have a point at the bottom about talking in the comments about anything just to encourage comments? Again, just an opinion more than anything. :+1: otherwise

**Comment by hpoom** [comment]
Good post, feels like it needs a close. A summary or wrap up.

**Pull request merged**